internal/services/appconfiguration/app_configuration.go

```diff
@@ -30,6 +30,9 @@ func appConfigurationGetKeyRefreshFunc(ctx context.Context, client *appconfigura
 				if utils.ResponseWasForbidden(autorest.Response{Response: v.Response}) {
 					return "Forbidden", "Forbidden", nil
 				}
+				if utils.ResponseWasNotFound(autorest.Response{Response: v.Response}) {
```

Comment:
Can we make this change above as well for `response.WasForbidden`
```suggestion
				if response.WasNotFound(v.Response) {
```